flatten-vs-shade incompatibility

When using Flatten Maven Plugin in the same project with Maven Shade Plugin then the dependency reduced POM is *not* flattened.

One of both plugins must have a bug, but which one?

I noticed that the cause seems to be this line of the shade plugin:
[`Model model = project.getOriginalModel();`](https://github.com/apache/maven-shade-plugin/blob/e1f181f9ef9f196f299abe4cceeb1caeefcf4cdc/src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java#L931)
For test purposes I changed it to `project.getModel()` which fixes the problem (i. e. it now reduces the *flattened* POM), but imposes a new problem: This does not flatten the POM, it flattens the *effective* POM.

Any ideas appreciated, so I could track the *original* cause, and file the bug in the *right* tracker.

I encountered the exact issue you described. 
I was able to fix this only by simply setting createDependencyReducedPom to false

Changing the order of execution between shade and flatten didn't work for me

The reduced pom is indeed not flattened but also I have noticed that there is a error with the generated reduced pom with  the parents's relativePath 

originally its 	<relativePath>../weasis-parent/pom.xml</relativePath>
and it becomes 	<relativePath>../../weasis-parent/pom.xml</relativePath> when reduced 

@see https://i.stack.imgur.com/boolA.png  

that's probably the reason why it cannot resolve the parents pom 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.2.4:shade (default) on project weasis-launcher: Error creating shaded jar: 1 problem was encountered while building the effective model for org.weasis.launcher:weasis-launcher:[unknown-version]
[ERROR] [FATAL] Non-resolvable parent POM for org.weasis.launcher:weasis-launcher:[unknown-version]: Could not transfer artifact org.weasis:weasis-parent:pom:${revision}${changelist} from/to hug-nexus-repo-proxy (http://..../repository/maven-public): Failed to transfer file: http://..../repository/maven-public/org/weasis/weasis-parent/$%7Brevision%7D$%7Bchangelist%7D/weasis-parent-$%7Brevision%7D$%7Bchangelist%7D.pom. Return code is: 400 , ReasonPhrase:Invalid repository path. and 'parent.relativePath' points at wrong local POM @ line 3, column 11
[ERROR]  for project org.weasis.launcher:weasis-launcher:[unknown-version] at ....
```

--- a/src/it/projects/flatten-shaded-drp/invoker.properties
+++ b/src/it/projects/flatten-shaded-drp/invoker.properties
@@ -1,0 +1,1 @@
+invoker.goals=package

--- a/src/it/projects/flatten-shaded-drp/pom.xml
+++ b/src/it/projects/flatten-shaded-drp/pom.xml
@@ -1,0 +1,105 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.codehaus.mojo.flatten.its</groupId>
+    <artifactId>flatten-shaded-drp</artifactId>
+    <version>0.0.1${rev}-SNAPSHOT</version>
+
+    <properties>
+        <flatten.its.version>3.2.1</flatten.its.version>
+        <rev></rev>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>16.0.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-archiver</artifactId>
+            <version>2.5</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <defaultGoal>package</defaultGoal>
+
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>flatten-maven-plugin</artifactId>
+                <configuration>
+                    <updatePomFile>true</updatePomFile>
+                    <flattenMode>ossrh</flattenMode>
+                    <embedBuildProfileDependencies>true</embedBuildProfileDependencies>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>flatten</id>
+                        <phase>prepare-package</phase>
+                        <goals>
+                            <goal>flatten</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>3.3.0</version>
+                <configuration>
+                    <minimizeJar>true</minimizeJar>
+                    <createDependencyReducedPom>true</createDependencyReducedPom>
+                    <artifactSet>
+                        <includes>
+                            <include>com.google.guava:*</include>
+                        </includes>
+                    </artifactSet>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>shade</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <!-- should be removed -->
+    <profiles>
+        <profile>
+            <id>foo</id>
+            <activation>
+                <property>
+                    <name>willNotBeActive</name>
+                </property>
+            </activation>
+
+            <dependencies>
+                <dependency>
+                    <groupId>org.codehaus.mojo.flatten.its</groupId>
+                    <artifactId>core</artifactId>
+                    <version>${flatten.its.version}</version>
+                </dependency>
+            </dependencies>
+        </profile>
+    </profiles>
+
+    <pluginRepositories>
+        <pluginRepository>
+            <id>no-one</id>
+            <url>@repository.proxy.url@</url>
+        </pluginRepository>
+    </pluginRepositories>
+
+
+</project>

--- a/src/it/projects/flatten-shaded-drp/src/main/java/com/example/Main.java
+++ b/src/it/projects/flatten-shaded-drp/src/main/java/com/example/Main.java
@@ -1,0 +1,12 @@
+package com.example;
+
+import java.util.List;
+import com.google.common.collect.ImmutableList;
+import org.apache.maven.archiver.MavenArchiver;
+
+public class Main {
+    public static void main(String[] args) {
+        List<Object> l = ImmutableList.of();
+        MavenArchiver a = new MavenArchiver();
+    }
+}

--- a/src/it/projects/flatten-shaded-drp/verify.groovy
+++ b/src/it/projects/flatten-shaded-drp/verify.groovy
@@ -1,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+File originalPom = new File( basedir, 'pom.xml' )
+assert originalPom.exists()
+
+def originalProject = new XmlSlurper().parse( originalPom )
+assert 1 ==  originalProject.dependencies.size()
+assert 2 ==  originalProject.dependencies.dependency.size()
+
+File flattendPom = new File( basedir, 'dependency-reduced-pom.xml' )
+assert flattendPom.exists()
+
+def flattendProject = new XmlSlurper().parse( flattendPom )
+assert 1 ==  flattendProject.dependencies.size()
+assert 1 ==  flattendProject.dependencies.dependency.size()
+
+def archiver = flattendProject.dependencies.dependency.find {
+    it.groupId == 'org.apache.maven' && it.artifactId == 'maven-archiver'
+}
+assert '2.5' == archiver.version.text()
+assert 'compile' == archiver.scope.text()
+
+assert 0 == flattendProject.build.size()
+assert 0 == flattendProject.profiles.size()
+assert 0 == flattendProject.pluginRepositories.size()

--- a/src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java
+++ b/src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java
@@ -422,6 +422,7 @@ public class FlattenMojo
         if ( isUpdatePomFile() )
         {
             this.project.setPomFile( flattenedPomFile );
+            this.project.setOriginalModel( flattenedPom );
         }
     }
 

